sessionctx/stmtctx/stmtctx.go

```diff
@@ -420,6 +425,55 @@ type StatementContext struct {
 	}
 }
 
+// NewStmtCtx creates a new statement context
+func NewStmtCtx() *StatementContext {
```

Comment:
I've tried to write a linter to make sure the `&StatementContext{}` is only constructed manually in `NewStmtCtx` or `NewStmtCtxWithTimeZone`. I'll submit that linter after this PR is merged 🍻